nyx_client/nyx_client/connection.py

```diff
@@ -0,0 +1,52 @@
+# Copyright © 2024 IOTIC LABS LTD. info@iotics.com
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://github.com/Iotic-Labs/nyx-sdk/blob/main/LICENSE
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for connection (3rd party data integrations) definitions in Nyx."""
+
+from dataclasses import dataclass
+from typing import Any
+
+
+@dataclass(frozen=True)
+class Connection:
+    """Represents a connection to a 3rd party data integration."""
+
+    id: str
+    """the id of the connection"""
+    name: str
+    """the name of the connection"""
+    json_blob: dict[str, Any]
```

Comment:
`json_blob` != dictionary - should it not say JSON blob? :thinking:  (If it's transmitted as an object rather than string)

Reply:
Sorry - I worded that badly. In the API it's a blob but not here.
If we call it "json_blob" everywhere it'll sound like users should provided a blob rather than dict and they have to decode it.
I meant should it e.g. be called fields/parameters/something-else ?